### DISCUSSION TITLE
Always open drawer and customization

### DISCUSSION THREAD

**Discussion by shyam-ayrshare** [chakra-ui/chakra-ui#9318] [Q&A] [answered]
There was a way in v2 involving variants where the user can interact with elements outside the drawer in the case that it's always open like in a side nav situation. I'm unable to get something like that working in v3. 

This is what I've tried and other variations of this: 
```
const drawerRecipe = defineRecipe({
  variants: {
    variant: {
      alwaysOpen: {
        dialog: {
          pointerEvents: "auto"
        }
      },
      custom: {
        dialog: {
          maxWidth: "400px"
        }
      }
    }
  }
});
```

Included it in the theme like so: 
```
const customConfig = defineConfig({
  theme: {
    recipes: {
      drawer: drawerRecipe
    }
  }
});

const config = mergeConfigs(defaultConfig, customConfig);
const system = createSystem(config);
```

And using it like so: 
``` 
<DrawerRoot
          trapFocus={false}
          preventScroll={false}
          open={true}
          closeOnEscape={false}
          closeOnInteractOutside={false}
          placement="left"
          variant="alwaysOpen"
   />
```

Also, was trying to have another drawer variant with a custom width (called custom above), but that is also not working. 

Has anybody been able to customize the drawer and/or get an always-open drawer working with pointer events enabled outside the drawer? Thanks!

**Comment by segunadebayo** [comment]
Can you share a reproduction? Use this to get started: https://stackblitz.com/edit/chakra-ui-v3

We intend to ship a SideNav or Sidebar component pretty soon so that should help.

**Reply by shyam-ayrshare** [reply]
https://stackblitz.com/edit/chakra-ui-v3-wrxodbum?file=src%2FApp.tsx

I'm not sure if I'm using the variant correctly in the drawer to begin with. Beyond that, wondering how the variant should be defined in the theme such that the user is still able to interact with the main content like hover, while the drawer is open. 

A Sidebar component would be awesome with this always-open option built-in. When is the component slated to be released?

Thanks.

**Comment by segunadebayo** [comment, marked as answer]
Drawer is a slot recipe, so `defineRecipe` never hits the parts. Use `defineSlotRecipe` and `slots: drawerAnatomy.keys()`.

For a persistent sidenav, set `modal={false}` `trapFocus={false}` `preventScroll={false}`, or use Sidebar. https://chakra-ui.com/docs/components/drawer
